gcc/testsuite/rust/execute/torture/builtin_macro_cfg.rs

```diff
@@ -12,7 +12,7 @@ extern "C" {
 fn print(s: &str) {
     unsafe {
         printf(
-            "%s\n\0" as *const str as *const i8,
+            "%s\0\n" as *const str as *const i8,
             s as *const str as *const i8,
         );
     }
```

Comment:
I'm still not sure that works as intended.  Per my understanding, libc `printf` upon parsing `%s` looks at the C-level string `s as *const str as *const i8`, and searches for NUL **in there**.

Reply:
Yes, sorry again, just doing multiple things at once :) It should now be fixed. The format string needs its own nul-terimnator, as well as each string given to printf as an argument, so I added extra \0 characters in both calls to `print` (which then calls `printf`)

Reply:
No worries.  :-)

ACK (conceptually) the NULs added at the `print` call sites.

But I still think changing the `printf` format string as you've currently got:

```diff
-"%s\n"
+"%s\n\0"
```

... is not necessary.

(I've not reviewed the `\r*` changes in detail.)

Reply:
> ... is not necessary.

Why not? We directly call into printf without adding a terminator without that change, and Rust strings are not nul-terminated. Am I missing something?

Reply:
Gah, you're right of course -- this is still a Rust string.  My bad, and sorry @CohenArthur for the confusion.

Reply:
No worries :D just making sure! Better safe than sorry :)